cirq-google/cirq_google/api/v2/program.py

```diff
@@ -59,8 +64,26 @@ def qubit_from_proto_id(proto_id: str) -> cirq.Qid:
     Returns:
         A `cirq.Qid` corresponding to the proto id.
     """
-    num_coords = len(proto_id.split('_'))
-    if num_coords == 2:
+    # Avoid circular import
+    from cirq_google.devices.coupler import Coupler
+
+    qubit_field = proto_id.split('_')
+    num_coords = len(qubit_field)
+    if proto_id[:2] == 'c_':
+        if num_coords == 5:
+            # 2 grid qubits: c_2_1_4_3
+            grid_qubit0_str = qubit_field[1] + '_' + qubit_field[2]
+            grid_qubit1_str = qubit_field[3] + '_' + qubit_field[4]
+            grid_qubit0 = grid_qubit_from_proto_id(grid_qubit0_str)
+            grid_qubit1 = grid_qubit_from_proto_id(grid_qubit1_str)
+            return Coupler(grid_qubit0, grid_qubit1)
+        elif num_coords == 3:
+            # 2 line qubits: c_2_4
+            # Or two named qubits: c_qubita_qubitb
+            line_qubit0 = qubit_from_proto_id(qubit_field[1])
+            line_qubit1 = qubit_from_proto_id(qubit_field[2])
+            return Coupler(line_qubit0, line_qubit1)
```

Comment:
Consider adding an `else` to `pass` like in the other conditional blocks, otherwise an invalid Coupler id will return as a `NamedQubit` which I don't think is the intended behavior, IIUC

Reply:
I am not sure I follow.

Adding
```
   else:
         pass
```
will have no effect since pass does nothing.  Also, falling back to `NamedQubit` seems reasonable if we cannot parse it as a coupler qubit.  That's what happens if you pass an unparseable grid qubit, such as '4a_4b'

Reply:
You're right, I read `pass` as `return` in this case. But actually it looks like there's a chance `grid_qubit_from_proto_id` and `qubit_from_proto_id` can throw a `ValueError`-- in that case you'll want to catch the exception right and fall back on `NamedQubit` instead of failing ?

Reply:
`grid_qubit_from_proto_id` and `line_qubit_from_proto_id` can throw exceptions.  `qubit_from_proto_id` does not.

Reply:
Line 95 and 96 use `grid_qubit_from_proto_id` does that need to caught?